x/supplier/module/tx_stake_supplier.go

```diff
@@ -45,9 +58,26 @@ $ pocketd tx supplier stake-supplier --config stake_config.yaml --keyring-backen
 				return err
 			}
 
+			signingKeyNameOrAddress, err := cmd.Flags().GetString(flags.FlagFrom)
+			if err != nil {
+				return err
+			}
+
 			// Ensure the --from flag is set before getting the client context.
-			if cmd.Flag(flags.FlagFrom) == nil {
-				if err = cmd.Flags().Set(flags.FlagFrom, supplierStakeConfigs.OwnerAddress); err != nil {
+			// Default to owner/operator signer in this order::
+			// 1. owner address - rationale: typical stake escrow source
+			// 2. operator address - rationale: --services-only, operator-custodial workflow, etc. use cases
+			if signingKeyNameOrAddress == "" {
+				switch {
+				case supplierStakeConfigs.OwnerAddress != "":
+					signingKeyNameOrAddress = supplierStakeConfigs.OwnerAddress
+				case supplierStakeConfigs.OperatorAddress != "":
+					signingKeyNameOrAddress = supplierStakeConfigs.OperatorAddress
+				default:
+					return types.ErrSupplierInvalidAddress.Wrap("no signer address provided")
```

Comment:
If we use a default signer, can you log something out so it tells the user what happened?